File: a_5_fragment/src/main/java/com/sj/a_5_fragment/MainActivity.java
package com.sj.a_5_fragment;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
    }

    private void initView() {
        getFragmentManager().beginTransaction().add(R.id.dynamic_fl,new DynamicFragment()).commit();
    }
}

File: navigation/src/main/java/com/sj/navigation/MainActivity.kt
package com.sj.navigation

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.navigation.NavController
import androidx.navigation.Navigation
import androidx.navigation.Navigation.findNavController
import androidx.navigation.ui.setupWithNavController
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        var navController = Navigation.findNavController(this, R.id.nav_host_fg)
        bnv.setupWithNavController(navController)
    }





    override fun onSupportNavigateUp() =
        findNavController(this, R.id.nav_host_fg).navigateUp()
}
